engine/eConstr.ml

```diff
@@ -651,6 +654,10 @@ let substnl subst n c = of_constr (Vars.substnl (List.map to_constr subst) n (to
 let substl subst c = of_constr (Vars.substl (List.map to_constr subst) (to_constr c))
 let subst1 c r = of_constr (Vars.subst1 (to_constr c) (to_constr r))
 
+let substnl_decl subst n d = of_rel_decl (Vars.substnl_decl (List.map to_constr subst) n (to_rel_decl d))
```

Comment:
Actually, to nitpick, I wonder why to call `unsafe_eq` unsafe. This data is an indirect way to say that the local `API` abstraction (indirectly) exports the constraint `t = Constr.constr` throughout the module `EConstr`. What can be unsafe in using it?